eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Genesis.hs

```diff
@@ -100,3 +125,211 @@ instance ToCBOR AlonzoGenesis where
           !> To maxValSize
           !> To collateralPercentage
           !> To maxCollateralInputs
+
+deriving instance ToJSON a => ToJSON (ExUnits' a)
+
+deriving instance FromJSON a => FromJSON (ExUnits' a)
+
+instance ToJSON ExUnits where
+  toJSON ExUnits {exUnitsMem = m, exUnitsSteps = s} =
+    object
+      [ "exUnitsMem" .= toJSON m,
+        "exUnitsSteps" .= toJSON s
+      ]
+
+instance FromJSON ExUnits where
+  parseJSON = Aeson.withObject "exUnits" $ \o -> do
+    mem <- o .: "exUnitsMem"
+    steps <- o .: "exUnitsSteps"
+    bmem <- checkWord64Bounds mem
+    bsteps <- checkWord64Bounds steps
+    return $ ExUnits bmem bsteps
+    where
+      checkWord64Bounds n =
+        if n >= fromIntegral (minBound @Word64)
+          && n <= fromIntegral (maxBound @Word64)
+          then pure n
+          else fail ("Unit out of bounds for Word64: " <> show n)
+
+toRationalJSON :: Rational -> Value
+toRationalJSON r =
+  case fromRationalRepetendLimited 20 r of
+    Right (s, Nothing) -> toJSON s
+    _ -> toJSON r
+
+instance ToJSON Prices where
+  toJSON Prices {prSteps, prMem} =
+    -- We cannot round-trip via NonNegativeInterval, so we go via Rational
+    object
+      [ "prSteps" .= toRationalJSON (BT.unboundRational prSteps),
+        "prMem" .= toRationalJSON (BT.unboundRational prMem)
+      ]
+
+instance FromJSON Prices where
+  parseJSON =
+    Aeson.withObject "prices" $ \o -> do
+      steps <- o .: "prSteps"
+      mem <- o .: "prMem"
+      prSteps <- checkBoundedRational steps
+      prMem <- checkBoundedRational mem
+      return Prices {prSteps, prMem}
+    where
+      -- We cannot round-trip via NonNegativeInterval, so we go via Rational
+      checkBoundedRational r =
+        case BT.boundRational r of
+          Nothing -> fail ("too much precision for bounded rational: " ++ show r)
+          Just s -> return s
+
+deriving newtype instance FromJSON CostModel
+
+deriving newtype instance ToJSON CostModel
+
+languageToText :: Language -> Text
+languageToText PlutusV1 = "PlutusV1"
+languageToText PlutusV2 = "PlutusV2"
+
+languageFromText :: MonadFail m => Text -> m Language
+languageFromText "PlutusV1" = pure PlutusV1
+languageFromText lang = fail $ "Error decoding Language: " ++ show lang
+
+instance FromJSON Language where
+  parseJSON = Aeson.withText "Language" languageFromText
+
+instance ToJSON Language where
+  toJSON = Aeson.String . languageToText
+
+instance ToJSONKey Language where
+  toJSONKey = toJSONKeyText languageToText
+
+instance FromJSONKey Language where
+  fromJSONKey = Aeson.FromJSONKeyTextParser languageFromText
+
+instance FromJSON AlonzoGenesis where
+  parseJSON = Aeson.withObject "Alonzo Genesis" $ \o -> do
+    coinsPerUTxOWord <-
+      o .: "lovelacePerUTxOWord"
+        <|> o .: "adaPerUTxOWord" --TODO: deprecate
```

Comment:
@Jimbo4350 - should I remove `adaPerUTxOWord`?

Reply:
Yeah it looks like we can remove this, there is no trace of it in the node repo.